alternator/conditions.cc

```diff
@@ -146,19 +175,13 @@ static bool check_BEGINS_WITH(const rjson::value* v1, const rjson::value& v2) {
     return val1.substr(0, val2.size()) == val2;
 }
 
-// Check if a JSON-encoded value equals any element of an array.
-static bool check_IN(const rjson::value* val, const rjson::value* array) {
-    if (!array || !array->IsArray()) {
-        throw api_error("ValidationException", "With ComparisonOperator, AttributeValueList must be given and an array");
-    }
-    if (array->Size() < 1) {
-        throw api_error("ValidationException", "IN operator requires nonempty AttributeValueList");
-    }
-    const auto& elem0 = (*array)[0];
-    if (!elem0.IsObject() || elem0.MemberCount() != 1) {
-        throw api_error("ValidationException", format("IN operator encountered malformed AttributeValue: {}", elem0));
+// Check if a JSON-encoded value equals any element of an array, which must have at least one element.
+static bool check_IN(const rjson::value* val, const rjson::value& array) {
+    if (!array[0].IsObject() || array[0].MemberCount() != 1) {
```

Comment:
verify_operand_count() already verifies this, so you don't need to verify it again here?

Reply:
This is subtly different: `verify_operand_count()` checks `array` but not `array[0]`.

Reply:
Oops, you're right 😂 I'll commit.